### PR TITLE
Customer object standardization

### PR DESCRIPTION
**Pull request by andykim**
## Description

- Customer object standardization
https://xendit.atlassian.net/browse/TPI-7484

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Cosmetic change (text changing or color changing)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code (if necessary), particularly in hard-to-understand areas
- [ ] My changes generate no new vulnerabilities
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshot

`put screenshot here if there are any ui changes`

### CODE REVIEW

File: Controller/Checkout/Invoice.php [hunk 1]
```diff
@@ -112,13 +104,7 @@ private function getApiRequestData($order)
...
             'success_redirect_url' => $this->getDataHelper()->getSuccessUrl(),
             'failure_redirect_url' => $this->getDataHelper()->getFailureUrl([$orderId]),
-            'customer' => (object)[
-                'given_names' => $order->getCustomerFirstname() ?: 'n/a',
-                'surname' => $order->getCustomerLastname() ?: 'n/a',
-                'email' => $order->getCustomerEmail() ?: 'noreply@mail.com',
-                'mobile_number' => $shippingAddress->getTelephone() ?: '',
-                'addresses' => [(object)$address]
-            ],
+            'customer' => $this->getDataHelper()->extractXenditInvoiceCustomerFromOrder($order),
```
Comment (candrasaputra) [inline Controller/Checkout/Invoice.php:107]: customer is optional.
if `$this->getDataHelper()->extractXenditInvoiceCustomerFromOrder($order)` return empty array.
is it mean we still send customer field?

File: Helper/Data.php [hunk 2]
```diff
@@ -559,4 +560,46 @@ public function truncateDecimal($amount)
...
+        $shippingAddress = $order->getShippingAddress();
+        $customerObject = [
+            'given_names' => $order->getCustomerFirstname(),
+            'surname' => $order->getCustomerLastname(),
+            'email' => $order->getCustomerEmail(),
+            'mobile_number' => $shippingAddress->getTelephone()
+        ];
+
+        $customerObject = array_filter($customerObject);
+        $customerObject['addresses'] = [
```
Comment (candrasaputra) [inline Helper/Data.php:579]: similar with above

File: Controller/Checkout/Invoice.php (hunk 1 above)
Comment (andykim) [reply Controller/Checkout/Invoice.php]: @candrasaputra I thought the empty customer object will be ignored on creating invoice service.
But yah, i updated it to check if empty and then not pass the customer.

File: Helper/Data.php (hunk 2 above)
Comment (andykim) [reply Helper/Data.php]: @candrasaputra Updated.

**Review by yanoandri — APPROVED** [review]

**Pull request merged**